assets/js/modules/analytics/components/module/ModuleOverviewWidget/index.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * ModuleOverviewWidget component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import { DATE_RANGE_OFFSET, MODULES_ANALYTICS } from '../../../datastore/constants';
+import Header from './Header';
+import Overview from './Overview';
+import SiteStats from './SiteStats';
+const { useSelect } = Data;
+
+export default function ModuleOverviewWidget( { Widget, WidgetReportError } ) {
+	const [ selectedStat, setSelectedState ] = useState( 0 );
+
+	const {
+		users,
+		sessions,
+		bounceRate,
+		avgSessionDuration,
+		loaded,
+		error,
+	} = useSelect( ( select ) => {
+		const {
+			startDate,
+			endDate,
+			compareStartDate,
+			compareEndDate,
+		} = select( CORE_USER ).getDateRangeDates( {
+			compare: true,
+			offsetDays: DATE_RANGE_OFFSET,
+		} );
+
+		return [ 'users', 'sessions', 'bounceRate', 'avgSessionDuration' ].reduce(
+			( acc, metric ) => {
+				const args = {
+					startDate,
+					endDate,
+					compareStartDate,
+					compareEndDate,
+					dimensions: 'ga:date',
+					metrics: [ `ga:${ metric }` ],
+				};
+
+				return {
+					...acc,
+					[ metric ]: select( MODULES_ANALYTICS ).getReport( args ),
+					loaded: acc.loaded && select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [ args ] ),
+					error: acc.error || select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [ args ] ),
+				};
+			},
+			{ loaded: true },
+		);
+	} );
```

Comment:
This should be queried in a single report. The dimensions and dates etc that we need are the same for every metric, so we can just query it all at once.

The overall goal is to keep reports specific to the widget they are used in. We shouldn't combine reports between widgets for less bandwidth, but a single widget should ensure its report is queried in an optimal way. Instead of having 4 API queries, this can become a single one.

Reply:
Updated.